zeton/api/user.py

```diff
@@ -77,4 +77,27 @@ def set_firstname():
         flash('Wprowadź imię')
     flash('Imię zostało zmienione')
 
-    return redirect(url_for('views.firstname_change'))
\ No newline at end of file
+    return redirect(url_for('views.firstname_change'))
+
+
+@bp.route('/<child_id>/assign/set_caregiver_to_child', methods=['POST'])
+@auth.login_required
+@auth.caregiver_only
+def set_caregiver_to_child(child_id):
+    child_id = int(child_id)
+
+    caregiver_username_to_child = request.form.get('caregiver_username_to_child')
+    caregiver_id_to_child = users.get_user_id(caregiver_username_to_child)
+    caregiver_data_to_child = users.get_user_data(caregiver_id_to_child)
+
+    if caregiver_id_to_child:
+        if caregiver_data_to_child['role'] == g.user_data['role']:
```

Comment:
Zmieniłbym ten fragment. Jeśli ten kod miał zabezpieczyć przed nieuprawnionym dostępem to dużo załatwia nam dekorator `@auth.caregiver_only` Do tego endpointu na pewno dostanie się jedynie `caregiver`.  Natomiast jeśli chcemy sprawdzić czy ktoś próbuje przypisać dziecko jako opiekuna to proponuje napisać to wprost. 
`if caregiver_data_to_child['role'] == 'caregiver'`  albo 
`if caregiver_data_to_child['role'] == 'child'`
Dodatkowo we flashu brakuje właśnie tego przypadku, jak chcemy przypisać dziecko jako opiekun drugiego dziecka. Aplikacja poprawnie się zachowuje ale powinna dać infromację zwrotną ` flash('Wprowadź poprawny login (username) opiekuna')`

Reply:
Prawda, nie powinno być możliwości przypisania dziecka do dziecka. Zrobione! :)